cmd/get/matchable_task_resource_attribute.go

```diff
@@ -0,0 +1,101 @@
+package get
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flyteorg/flytectl/cmd/config"
+	"github.com/flyteorg/flytectl/cmd/config/subcommand"
+	cmdCore "github.com/flyteorg/flytectl/cmd/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+)
+
+const (
+	taskResourceAttributesShort = "Gets matchable resources of task attributes"
+	taskResourceAttributesLong  = `
+Retrieves task  resource attributes for given project,domain combination or additionally with workflow name.
+
+Retrieves task resource attribute for project and domain
+Here the command get task resource attributes for  project flytectldemo and development domain.
+::
+
+ flytectl get task-resource-attribute -p flytectldemo -d development 
+
+eg : O/P
+
+.. code-block:: json
+
+ {"Project":"flytectldemo","Domain":"development","Workflow":"","defaults":{"cpu":"1","memory":"150Mi"},"limits":{"cpu":"2","memory":"450Mi"}}
+
+Writing the task resource attribute to a file. If there are no task resource attributes a file would be written with basic data populated.
+Here the command gets task resource attributes and writes the config file to tra.yaml
+eg:  content of tra.yaml
+
+::
+
+ flytectl get task-resource-attribute --attrFile tra.yaml
+
+
+.. code-block:: yaml
+
+	Domain: development
+	Project: flytectldemo
+	Workflow: ""
+	defaults:
+	  cpu: "1"
+	  memory: 150Mi
+	limits:
+	  cpu: "2"
+	  memory: 450Mi
+
+Usage
+`
+)
+
+func getTaskResourceAttributes(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
+	var project string
+	var domain string
+	var workflowName string
+
+	// Get the project domain workflow name parameters from the command line. Project and domain are mandatory for this command
+	project = config.GetConfig().Project
+	domain = config.GetConfig().Domain
+	if len(args) == 1 {
+		workflowName = args[0]
+	}
+	// Construct a shadow config for TaskResourceAttribute. The shadow config is not using ProjectDomainAttribute/Workflowattribute directly inorder to simplify the inputs.
+	taskResourceAttrFileConfig := subcommand.TaskResourceAttrFileConfig{Project: project, Domain: domain, Workflow: workflowName}
+	// Get the attribute file name from the command line config
+	fileName := subcommand.DefaultTaskResourceFetchConfig.AttrFile
+
+	if len(workflowName) > 0 {
+		// Fetch the workflow attribute from the admin
+		workflowAttr, err := cmdCtx.AdminFetcherExt().FetchWorkflowAttributes(ctx,
+			project, domain, workflowName, admin.MatchableResource_TASK_RESOURCE)
+		if err != nil {
+			taskResourceAttrFileConfig.DumpTaskResourceAttr(ctx, fileName)
```

Comment:
why dump the resources attributes if the get call returned an error? won't they be empty anyways?

Reply:
So my earlier version of the code would dump the keys even if the values didnt exist.
That way the user doesn't have to refer to structure from somewhere else.

And also he can modify the ones which he wants to be updated and leave the rest as empty .
Now that i am relying on directly idl structs and which rely on pointers , the Marshaller skips them since omiempty is controlled in the idl structs.

I will remove this as it might get confusing and will have the users copy the structure from the doc